### PR TITLE
quill-cpp: init at 3.8.0

### PR DESCRIPTION
**Pull request by vaci**
## Description of changes

Quill is a high-performance, cross-platform logging library designed for C++14 and onwards.
https://github.com/odygrd/quill

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/qu/quill-cpp/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  meta = with lib; {
+    description = "A high-performance, cross-platform logging library";
+    homepage = "https://github.com/odygrd/quill";
+    license = licenses.mit;
+    maintainers = [ maintainers.vaci ];
+    sourceProvenance = [
```
Comment (Aleksanaa) [inline pkgs/by-name/qu/quill-cpp/package.nix:28]: Not needed

**Review by Aleksanaa — APPROVED** [review]
Others lgtm

File: pkgs/by-name/qu/quill-cpp/package.nix (hunk 1 above)
Comment (vaci) [reply pkgs/by-name/qu/quill-cpp/package.nix:28]: Can any/of these metadata be omitted, then?

Comment (Aleksanaa) [reply pkgs/by-name/qu/quill-cpp/package.nix:28]: > The presence of a particular sourceType in a package’s meta.sourceProvenance list indicates that the package contains some components falling into that category, though the absence of that sourceType does not guarantee the absence of that category of sourceType in the package’s contents. A package with no meta.sourceProvenance set implies it has no known sourceTypes other than fromSource.

In https://nixos.org/manual/nixpkgs/unstable/#lib.sourceTypes.fromSource

Comment (vaci) [reply pkgs/by-name/qu/quill-cpp/package.nix:28]: I see, thanks for the explanation!

**Comment by SuperSandro2000** [comment]
dupe of #316404

**Pull request closed without merging**